### DISCUSSION TITLE
Did I get this right?

### DISCUSSION THREAD

**Discussion by beppe9000** [community/community#22569] [New to GitHub]
<p>I want to fork a third party project in my private repository, store my new modifications in a “dev” branch and keep the repository up to date with the upstream commits.</p>
<p>Setup</p>
<ol>
<li>git clone &lt;url&gt;</li>
<li>git branch dev</li>
<li>git checkout dev</li>
</ol>
<p>Update</p>
<ol>
<li>git fetch</li>
<li>git checkout dev</li>
<li>git rebase upstream/master</li>
</ol>
<p>Is this correct?</p>

**Comment by AndreaGriffiths11** [comment, marked as answer]
<p>HI <a class="mention" href="/beppe9000">@beppe9000</a>,</p>
<p>Thanks for being here! Yes, you are correct to keep your fork up to date:</p>
<pre><code>git fetch upstream

git rebase upstream/master
</code></pre>

**Comment by beppe9000** [comment]
<p>thanks so the upstream parameter was needed</p>
